### ISSUE TITLE
Make tables list configurable at compile time

### ISSUE THREAD

**Issue by anatol** [osquery/osquery#6478]
I am looking at osquery/tables/system/CMakeLists.txt file and see that all the tables are compiled unconditionally:

```
      linux/deb_packages.cpp
      linux/disk_encryption.cpp
      linux/elf_info.cpp
      linux/groups.cpp
      linux/intel_me.cpp
      linux/kernel_info.cpp
      linux/kernel_modules.cpp
      linux/md_tables.cpp
      linux/memory_info.cpp
      linux/memory_map.cpp
      linux/model_specific_register.cpp
      linux/mounts.cpp
      linux/npm_packages.cpp
      linux/os_version.cpp
      linux/pacman_packages.cpp
      linux/pci_devices.cpp
      linux/portage.cpp
      linux/process_open_files.cpp
      linux/process_open_pipes.cpp
      linux/processes.cpp
      linux/rpm_packages.cpp
```

It would be great to make some of these tables configurable at compile-time. For example if I plan to run `osquery` at Debian I do not really need RPM support and its dependencies. Thus having something like `cmake -DOSQUERY_TABLE_RPM=OFF` would be useful.

**Comment by theopolis** [comment]
I agree that configure-time feature flags would be a helpful addition. These can map to some of the "heavy" dependencies and plugins like AWS, TSK, and like you mentioned, flavor specific support.

I do not think there is anything stopping us now. Roughly:
- Agree on a convention, for example: `-DOSQUERY_FEATURE_$NAME`.
- Adopt this in both the CMake code that selects our code and the choice to Find the library.
- Discuss how to handle this in testing.

The last point is interesting. I do not think we'll be able to support the matrix of turning on/off these feature flags within our CI. This means there might be hidden regressions and incompatibilities that are not discovered per-PR.
